Indicators/Gamma.cs

```diff
@@ -0,0 +1,383 @@
+        {
+        }
+
+        // Calculate the Gamma of the option
+        protected override decimal CalculateGreek(DateTime time)
+        {
+            var timeToExpiration = Convert.ToDecimal((Expiry - time).TotalDays) / 365m;
+            var math = OptionGreekIndicatorsHelper.DecimalMath;
+
+            switch (_optionModel)
+            {
+                case OptionPricingModelType.BlackScholes:
+                    var norm = new Normal();
+                    var d1 = OptionGreekIndicatorsHelper.CalculateD1(UnderlyingPrice, Strike, timeToExpiration, RiskFreeRate, DividendYield, ImpliedVolatility);
+
+                    // allow at least 1% IV
+                    return math(norm.Density, -d1) / UnderlyingPrice / Math.Max(ImpliedVolatility, 0.01m) / math(Math.Sqrt, timeToExpiration);
+
+                case OptionPricingModelType.BinomialCoxRossRubinstein:
+                case OptionPricingModelType.ForwardTree:
+                    var upFactor = math(Math.Exp, ImpliedVolatility * math(Math.Sqrt, timeToExpiration / 200));
+                    if (upFactor == 1)
+                    {
+                        // provide a small step to estimate gamma
+                        upFactor = 1.0001m;
+                    }
+
+                    // Finite differncing approach
+                    var sU = UnderlyingPrice * upFactor * upFactor;
+                    var sD = UnderlyingPrice / upFactor / upFactor;
+
+                    var fU = 0m;
+                    var fM = 0m;
+                    var fD = 0m;
+                    if (_optionModel == OptionPricingModelType.BinomialCoxRossRubinstein)
+                    {
+                        fU = OptionGreekIndicatorsHelper.CRRTheoreticalPrice(ImpliedVolatility, sU, Strike, timeToExpiration, RiskFreeRate, DividendYield, Right);
+                        fM = OptionGreekIndicatorsHelper.CRRTheoreticalPrice(ImpliedVolatility, UnderlyingPrice, Strike, timeToExpiration, RiskFreeRate, DividendYield, Right);
+                        fD = OptionGreekIndicatorsHelper.CRRTheoreticalPrice(ImpliedVolatility, sD, Strike, timeToExpiration, RiskFreeRate, DividendYield, Right);
+                    }
+                    else if (_optionModel == OptionPricingModelType.ForwardTree)
+                    {
+                        fU = OptionGreekIndicatorsHelper.ForwardTreeTheoreticalPrice(ImpliedVolatility, sU, Strike, timeToExpiration, RiskFreeRate, DividendYield, Right);
+                        fM = OptionGreekIndicatorsHelper.ForwardTreeTheoreticalPrice(ImpliedVolatility, UnderlyingPrice, Strike, timeToExpiration, RiskFreeRate, DividendYield, Right);
+                        fD = OptionGreekIndicatorsHelper.ForwardTreeTheoreticalPrice(ImpliedVolatility, sD, Strike, timeToExpiration, RiskFreeRate, DividendYield, Right);
+                    }
+
+                    var gammaU = (fU - fM) / (sU - UnderlyingPrice);
+                    var gammaD = (fM - fD) / (UnderlyingPrice - sD);
```

Comment:
There are a few potential division by zero cases in this method we should protect against? We have a `SafeDivision` extension method we could use

Reply:
I don't think so?
```
if (upFactor == 1)
{
    // provide a small step to estimate gamma
    upFactor = 1.0001m;
}

 // Finite differncing approach
var sU = UnderlyingPrice * upFactor * upFactor;
var sD = UnderlyingPrice / upFactor / upFactor;
```
`upFactor` will never be 1.0, so `sU` & `sD` will never equal `UnderlyingPrice` and the denominator will never be 0m.